src/reaktoro_pse/tutorials/softening_acid_ro_example.ipynb

```diff
@@ -0,0 +1,1324 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "2025-08-28 14:49:58 [INFO] idaes.reaktoro_pse.core.reaktoro_inputs: Exact speciation is not provided! Fixing aqueous solvent and, excluding H\n",
+      "2025-08-28 14:49:58 [INFO] idaes.reaktoro_pse.core.reaktoro_inputs: Exact speciation is not provided! Fixing aqueous solvent and, excluding O\n",
+      "2025-08-28 14:49:58 [INFO] idaes.reaktoro_pse.core.reaktoro_solver: rktSolver inputs: ['[Cl]', '[C]', '[Na]', '[Mg]', '[S]', '[K]', '[Ca]', '[H2O]', '[H]', '[O]', '[H+]']\n",
+      "2025-08-28 14:49:58 [INFO] idaes.reaktoro_pse.core.reaktoro_solver: rktSolver constraints: ['charge', 'C_constraint', 'Na_constraint', 'Mg_constraint', 'S_constraint', 'K_constraint', 'Ca_constraint', 'H2O_constraint', 'H_dummy_constraint', 'O_dummy_constraint', 'pH']\n",
+      "2025-08-28 14:49:58 [INFO] idaes.reaktoro_pse.core.reaktoro_solver: rktSolver inputs: ['[H+]', '[H]', '[C]', '[O]', '[Na]', '[Mg]', '[S]', '[Cl]', '[K]', '[Ca]']\n",
+      "2025-08-28 14:49:58 [INFO] idaes.reaktoro_pse.core.reaktoro_solver: rktSolver constraints: ['charge', 'H_constraint', 'C_constraint', 'O_constraint', 'Na_constraint', 'Mg_constraint', 'S_constraint', 'Cl_constraint', 'K_constraint', 'Ca_constraint']\n",
+      "2025-08-28 14:49:58 [INFO] idaes.reaktoro_pse.core.reaktoro_gray_box: RKT gray box using LBFGS hessian type\n"
+     ]
+    }
+   ],
+   "source": [
+    "m.fs.softening.eq_precipitation = ReaktoroBlock(\n",
+    "    aqueous_phase={\n",
+    "        \"composition\": m.fs.feed.species_mass_flow,  # This is the speciesmass flow\n",
+    "        \"convert_to_rkt_species\": True,  # We can use default converter as its defined for default database (Phreeqc and pitzer)\n",
+    "        \"activity_model\": reaktoro.ActivityModelPitzer(),  # Can provide a string, or Reaktoro initialized class\n",
+    "        \"fixed_solvent_specie\": \"H2O\",  # We need to define our aqueous solvent as we have to speciate the block\n",
+    "    },\n",
+    "    system_state={\n",
+    "        \"temperature\": m.fs.feed.temperature,\n",
+    "        \"pressure\": m.fs.feed.pressure,\n",
+    "        \"pH\": m.fs.feed.pH,\n",
+    "    },\n",
+    "    mineral_phase={\"phase_components\": [\"Calcite\"]},\n",
+    "    chemistry_modifier={\n",
+    "        \"CaO\": m.fs.softening.reagents_mol_flow[\"CaO\"],\n",
+    "        \"Na2CO3\": m.fs.softening.reagents_mol_flow[\"Na2CO3\"],\n",
+    "    },\n",
+    "    outputs=m.fs.softening.reaktoro_outputs,  # This is the output dictionary we defined above\n",
+    "    database_file=\"pitzer.dat\",  # needs to be a string that names the database file or points to its location\n",
+    "    build_speciation_block=True,  # We provided apparent species so we need to speciate them.\n",
+    ")"
+   ]
+  },
```

Comment:
Perhaps these should also be aggregated with the above cells to the extent that you deem tolerable. 

Another thought here (if timing is an issue) is that we can have a "skinny" version of this demo and a "detailed" version, which is what some other presenters may be doing. 

We could present the "skinny version" (where much of these details are aggregated as suggested, or even packed into higher level functions that we import to shield the audience from supersaturation, pun intended) and then have this copy of the notebook as the "detailed" demo that we refer people to after the session to learn more of the nitty gritty details and go through all the explanation at their own pace.